### PR TITLE
Port Random Bark System

### PR DESCRIPTION
**Pull request by VMSolidus**
# Description

https://github.com/Simple-Station/Parkstation-Friendly-Chainsaw/pull/11

Later, I would like to have this system not just for animals, but for implementations of humanoid NPCs, since it can just as easily be used for human NPCs.
I could make space bandits that CHEEKI BREEKI at people.

# Changelog

:cl: DEATHB4DEFEAT
- add: Animals will no longer be a silent, soulless shell

### CODE REVIEW

File: Content.Server/Speech/Components/RandomBarkComponent.cs
Comment (DEATHB4DEFEAT) [inline Content.Server/Speech/Components/RandomBarkComponent.cs:1]: Make these comments xmldocs

File: Content.Server/Speech/Systems/RandomBarkSystem.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+using Content.Server.Chat.Systems;
+using Content.Shared.Mind.Components;
+using Robust.Shared.Random;
+using Content.Server.Speech.Components;
+
+namespace Content.Server.SimpleStation14.Speech.Systems;
```
Comment (DEATHB4DEFEAT) [inline Content.Server/Speech/Systems/RandomBarkSystem.cs:6]:
```suggestion
namespace Content.Server.Speech.Systems;
```

**Review by DEATHB4DEFEAT — CHANGES_REQUESTED** [review]

**Review by DEATHB4DEFEAT — APPROVED** [review]

File: Resources/Prototypes/Entities/Mobs/NPCs/simplemob.yml [hunk 2]
```diff
@@ -104,3 +104,4 @@
   - type: MobPrice
     price: 150
   - type: FloatingVisuals
+  - type: Speech
```
Comment (DangerRevolution) [inline Resources/Prototypes/Entities/Mobs/NPCs/simplemob.yml:107]: What does this do?

File: Content.Server/Speech/Components/RandomBarkComponent.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    ///     Accumulator for counting time since the last bark
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    public float BarkAccumulator = 8f;
+
+    /// <summary>
+    ///     Multiplier applied to the random time. Good for changing the frequency without having to specify exact values
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    public float BarkMultiplier = 1f;
```
Comment (DangerRevolution) [inline Content.Server/Speech/Components/RandomBarkComponent.cs:37]: Just to clarify; what does this do? Not really getting from the description

File: Resources/Prototypes/Entities/Mobs/NPCs/simplemob.yml (hunk 2 above)
Comment (VMSolidus) [reply Resources/Prototypes/Entities/Mobs/NPCs/simplemob.yml:107]: Needed to make the barks in the first place.

File: Content.Server/Speech/Components/RandomBarkComponent.cs (hunk 3 above)
Comment (VMSolidus) [reply Content.Server/Speech/Components/RandomBarkComponent.cs:37]: image
It accelerates the rate at which barks accumulate. The higher the multiplier is, the more often the barks happen. Make it lower to make them bark less often.

File: Content.Server/Speech/Components/RandomBarkComponent.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+/// <summary>
+///     Sends a random message from a list with a provided min/max time.
+/// </summary>
+[RegisterComponent]
+public sealed partial class RandomBarkComponent : Component
+{
+    /// <summary>
+    ///     Should the message be sent to the chat log?
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
```
Comment (DEATHB4DEFEAT) [inline Content.Server/Speech/Components/RandomBarkComponent.cs:12]:
```suggestion
    [DataField]
```

File: Content.Server/Speech/Components/RandomBarkComponent.cs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    public bool ChatLog = false;
+
+    /// <summary>
+    ///     Minimum time an animal will go without speaking
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
```
Comment (DEATHB4DEFEAT) [inline Content.Server/Speech/Components/RandomBarkComponent.cs:18]:
```suggestion
    [DataField]
```

File: Content.Server/Speech/Components/RandomBarkComponent.cs [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    public int MinTime = 45;
+
+    /// <summary>
+    ///     Maximum time an animal will go without speaking
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
```
Comment (DEATHB4DEFEAT) [inline Content.Server/Speech/Components/RandomBarkComponent.cs:24]:
```suggestion
    [DataField]
```

File: Content.Server/Speech/Components/RandomBarkComponent.cs [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    public int MaxTime = 350;
+
+    /// <summary>
+    ///     Accumulator for counting time since the last bark
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
```
Comment (DEATHB4DEFEAT) [inline Content.Server/Speech/Components/RandomBarkComponent.cs:30]:
```suggestion
    [DataField]
```

File: Content.Server/Speech/Components/RandomBarkComponent.cs [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    public float BarkAccumulator = 8f;
+
+    /// <summary>
+    ///     Multiplier applied to the random time. Good for changing the frequency without having to specify exact values
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
```
Comment (DEATHB4DEFEAT) [inline Content.Server/Speech/Components/RandomBarkComponent.cs:36]:
```suggestion
    [DataField]
```

File: Content.Server/Speech/Components/RandomBarkComponent.cs [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    public float BarkMultiplier = 1f;
+
+    /// <summary>
+    ///     List of things to be said. Filled with garbage to be modified by an accent, but can be specified in the .yml
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
```
Comment (DEATHB4DEFEAT) [inline Content.Server/Speech/Components/RandomBarkComponent.cs:42]:
```suggestion
    [DataField]
```

**Review by DangerRevolution — COMMENTED** [review]
Does this output using their language? I'm assuming it does...

Also; this is going to be insanely pedantic but I'd appreciate the changelog entry being changed to give more context as to what the actual change is, for when someone looks at the entries lol

**Comment by DEATHB4DEFEAT** [comment]
> Does this output using their language? I'm assuming it does...

It should.

> Also; this is going to be insanely pedantic but I'd appreciate the changelog entry being changed to give more context as to what the actual change is, for when someone looks at the entries lol

The changelog is perfect.

**Review by DangerRevolution — APPROVED** [review]
Good enough for death; good enough for me

**Pull request merged**